### DISCUSSION TITLE
请问wsl2中的neovim如何获取宿主机win的剪切板内容？

### DISCUSSION THREAD

**Discussion by allworldg** [nshen/learn-neovim-lua#55] [Q&A] [answered]
Version: neovim 0.7.2 , host win11 ,wsl2 Ubuntu20.02, 终端：windows terminal
我想实现在windows 复制内容后在neovim里按 `p` 可以粘贴，在neovim按yy或者`ctrl+c`复制后可以在windows 粘贴。
尝试：
1. 设置vim.opt.clipboard = ’unnamedplus' 
2. 下载安装`win32yank`. 
![image](https://user-images.githubusercontent.com/48782874/178500802-f3ef4f1d-bcfe-4c95-a6a2-d6c41ee8d91b.png)
![image](https://user-images.githubusercontent.com/48782874/178501286-e3efc6d3-e6c2-4fcc-b932-812855337f86.png)

3. 按p粘贴提示 
![image](https://user-images.githubusercontent.com/48782874/178503294-ce85d64d-aafd-4081-9c4c-264e54d8c178.png)

结果：只能在windows复制后ctrl+v粘贴到neovim，wsl2内部neovim无论用yy还是ctrl+c都无法使用复制。

**Comment by nshen** [comment]
翻了一下我之前在知乎写的，参考一下：

官方给的解答是这样的

How to use the Windows clipboard from WSL?
https://​github.com/neovim/neovim/wiki/FAQ#how-to-use-the-windows-clipboard-from-wsl

首先确认有 win32yank.exe 在正确的目录里

![image](https://user-images.githubusercontent.com/181506/178515890-c97f5227-33be-4715-9f87-7224821841f1.png)


如果没有就下载一个

```
curl -sLo/tmp/win32yank.zip https://github.com/equalsraf/win32yank/releases/download/v0.0.4/win32yank-x64.zip
unzip -p /tmp/win32yank.zip win32yank.exe > /tmp/win32yank.exe
chmod +x /tmp/win32yank.exe
sudo mv /tmp/win32yank.exe /usr/local/bin/
```

然后在 init.vim 里添加

`set clipboard=unnamedplus`

表示使用系统剪切板。

理论上重启vim，以后 y 和 p 就自动使用系统的剪切板了。

现实是我之前设置的不好用，会卡住vim，可能是windows版本的原因。

```
版本 Windows 10 家庭中文版
版本号 21H1
安装日期 ‎2020/‎9/‎5
操作系统内部版本 19043.1387
体验 Windows Feature Experience Pack 120.2212.3920.0
```

你可以先试一下官方的方法，看看是不是跟我一样。也许你的版本就好用了，那就最好了。

然后是我找到的第 2 种方法：

把设置的 setclipboard 删掉

![image](https://user-images.githubusercontent.com/181506/178517143-4ef8dab4-a3b1-44f2-8623-f1304943987e.png)

使用 Terminal 里的粘贴快捷键。 可以把外边复制的文本粘贴到 nvim 里了

![v2-dafae0560d6e2fc05c347481f374bc3e_720w](https://user-images.githubusercontent.com/181506/178550416-b3fa51d8-e7c4-47d9-8cb4-ec058c4c809c.gif)


如果想从Nvim里把东西复制到外边，需要在init.vim 里添加一条自动命令

```
" 复制到windows剪贴板
" How to "copy to clipboard" in vim of Bash on Windows?
autocmd TextYankPost * if v:event.operator ==# 'y' | call system('/mnt/c/Windows/System32/clip.exe', @0) | endif
```

我之前测试大部分情况是OK的，希望你也可以。

**Reply by allworldg** [reply]
大佬的知乎帖子和官方我之前翻到过，我就是下了之后仍然无法共享windows剪切板。

**Reply by allworldg** [reply]
搞定了，感谢大佬

**Reply by nshen** [reply]
怎么搞定的，分享一下吧

**Reply by allworldg** [reply]
就是大佬你的第二个方法，和@kang8分享的差不多，调用clip复制粘贴。（win32yank实在搞不定，暂时放弃了）

**Comment by kang8** [comment, marked as answer]
试试这里的代码：

https://github.com/kang8/vimrc/blob/master/core/basic.vim#L61-L68

**Reply by allworldg** [reply]
正好在其他地方也看到这串代码，可以了，十分感谢！
